server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -215,7 +215,7 @@ public void collectSearchShards(IndicesOptions indicesOptions, String preference
                                     ActionListener<Map<String, ClusterSearchShardsResponse>> listener) {
         final CountDown responsesCountDown = new CountDown(remoteIndicesByCluster.size());
         final Map<String, ClusterSearchShardsResponse> searchShardsResponses = new ConcurrentHashMap<>();
-        final AtomicReference<TransportException> transportException = new AtomicReference<>();
+        final AtomicReference<RemoteTransportException> transportException = new AtomicReference<>();
```

Comment:
`RemoteTransportException` implements `ElasticsearchWrapperException`, then unwrapCause will unwrap the inner cause and status will return the inner status see https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/ElasticsearchException.java#L235

This behaviour is tricky. I think how ElasticsearchException#status works may cause problems when for instance it is thrown in the high-level REST client, I will have to dig deeper on that and probably open a follow-up PR, but I think this change is enough for cross-cluster search.